src/components/ReportActionItem/MoneyRequestPreview.js

```diff
@@ -156,6 +158,11 @@ function MoneyRequestPreview(props) {
     const hasFieldErrors = TransactionUtils.hasMissingSmartscanFields(props.transaction);
     const isDistanceRequest = TransactionUtils.isDistanceRequest(props.transaction);
 
+    // Show the merchant for IOUs and expenses only if they are custom or not related to scanning smartscan
+    const shouldShowMerchant =
+        !_.isEmpty(requestMerchant) && !props.isBillSplit && requestMerchant !== CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT && requestMerchant !== CONST.TRANSACTION.DEFAULT_MERCHANT;
+    const shouldShowDescription = !_.isEmpty(description) && !shouldShowMerchant;
```

Comment:
We should have checked if the receipt was not being scanned here. Not doing so caused https://github.com/Expensify/App/issues/31104.

Reply:
Okay, thank you for letting me know. AFAIK, the bug already fixed.